provider/terraform/resource_override.rb

```diff
@@ -19,13 +19,40 @@ class Terraform < Provider::AbstractCore
     # Collection of properties allowed in the ResourceOverride section for
     # Terraform. All properties should be `attr_reader :<property>`
     module OverrideProperties
+      # The Terraform resource id format used when calling #setId(...).
+      # For instance, `{{name}}` means the id will be the resource name.
+      attr_reader :id_format
+
+      attr_reader :documentation
+    end
+
+    # Terraform specific documentation additions
+    class Documentation < Api::Object
+      # Example Terraform configurations to include in documentation
+      attr_reader :examples
+
+      def validate
+        super
+
+        check_optional_property :examples, String
+      end
     end
 
     # A class to control overridden properties on terraform.yaml in lieu of
     # values from api.yaml.
     class ResourceOverride < Provider::ResourceOverride
       include OverrideProperties
 
+      def validate
+        super
+
+        @documentation ||= Provider::Terraform::Documentation.new
+
+        check_property :documentation, Provider::Terraform::Documentation
+
+        check_optional_property :id_format, String
```

Comment:
According to resource.erb:96 id_format is mandatory.

Reply:
Fixed with default value